File: lab_8/MyOop/Square.cpp
#include "Square.h"

const std::string Square::figureName = "Квадрат";

Square::Square(double side, const std::string& color) 
    : Rectangle(side, side, color) {}

std::string Square::getFigureName() const {
    return figureName;
}

std::string Square::repr() const {
    std::ostringstream oss;
    oss << std::fixed << std::setprecision(2);
    oss << "Фигура: " << figureName 
        << ", сторона: " << getSide() 
        << ", цвет: " << getColor() 
        << ", площадь: " << area();
    return oss.str();
}

File: lab_6/LabOop/Square.cs
using System;

namespace lab_6.LabOop {
    public class Square : Rectangle {
        public override string FigureName { get; } = "Квадрат";

        public Square(double side, string color) : base(side, side, color) {
        }

        public double Side {
            get { return Width; }
        }

        public override string Repr() {
            return $"Фигура: {FigureName}, сторона: {Side:F2}, цвет: {FigureColor}, площадь: {CalculateArea():F2}";
        }

        public override string ToString() {
            return Repr();
        }
    }
}